### PR TITLE
Proofread Oracle SELECT Hierarchical Query and GroupBy Clause

### PR DESCRIPTION
**Pull request by ThanoshanMV**
#10111 

Hi @wgy8283335, I've proofread Oracle [SELECT](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/SELECT.html#GUID-CFA006CA-6FF1-4972-821E-6996142A51C6) hierarchical query and group by clause. Please check it.

Changes proposed in this pull request:
- Proofread Oracle SELECT hierarchical query and group by clause.
- Add test cases for hierarchical query and group by clause.
- I created new constructors for `ColumnOrderByItem`, `ExpressionOrderByItem` and `IndexOrderByItem` without having `OrderDirection` as a construction parameter. Since Oracle `SELECT` group by clause is not associated with any order directions, I created those new constructors. I'll remove them if they aren't recommended.
- I added `ExpectedAggregationProjection` in `ExpectedExpression` and asserted it in `ExpressionAssert`.

### CODE REVIEW

**Comment by tristaZero** [comment]
Hi @ThanoshanMV 
What you did these days are pretty good to me. And @wgy8283335 still highly ranks your performance. 
The first evaluation of GSoC is coming, I will submit the evaluation ASAP and hope you can keep it up.
Have a nice day.

**Comment by wgy8283335** [comment]
@ThanoshanMV There are conflicts that need to be resovled.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4 [hunk 1]
```diff
@@ -622,8 +622,34 @@ whereClause
...
+    : GROUPING SETS LP_ (rollupCubeClause | groupingExprList) (COMMA_ (rollupCubeClause | groupingExprList))* RP_
+    ;
+
+groupingExprList
+    : expressionList (COMMA_ expressionList)*
+    ;
+
+expressionList
+    : exprs
+    | LP_ expr? (COMMA_ expr?)* RP_
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4:652]: Do you think this is better?
`exprs | LP_ expr? (COMMA_ expr)* RP_`

**Comment by wgy8283335** [comment]
> * I created new constructors for `ColumnOrderByItem`, `ExpressionOrderByItem` and `IndexOrderByItem` without having `OrderDirection` as a construction parameter. Since Oracle `SELECT` group by clause is not associated with any order directions, I created those new constructors. I'll remove them if they aren't recommended.

@tristaZero Do you think it is necessary to add constructor without having `OrderDirection`, as below:
```
    
    public IndexOrderByItemSegment(final int startIndex, final int stopIndex, final int columnIndex) {
        super(startIndex, stopIndex, OrderDirection.ASC, OrderDirection.ASC);
        this.columnIndex = columnIndex;
    }
```
We still could use this constructor to create IndexOrderByItemSegment object.
```
    public IndexOrderByItemSegment(final int startIndex, final int stopIndex, final int columnIndex, final OrderDirection orderDirection) {
        super(startIndex, stopIndex, orderDirection, OrderDirection.ASC);
        this.columnIndex = columnIndex;
    }
```

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4 (hunk 1 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4:652]: Yes, I'll change that.

**Comment by tristaZero** [comment]
Hi how is it going? @ThanoshanMV @wgy8283335

**Comment by ThanoshanMV** [comment]
Hi @tristaZero, 

@wgy8283335 reviewed this PR. We need to get insight from you whether is it okay on creating a new constructor for `ColumnOrderByItem`, `ExpressionOrderByItem` and `IndexOrderByItem` without having `OrderDirection` as a parameter. If it's not recommended, I'll remove it.

**Comment by wgy8283335** [comment]
@ThanoshanMV 
I think maybe we could not add the new constructor this time, right?

**Comment by ThanoshanMV** [comment]
Yes @wgy8283335, I'll change that and utilize the existing constructor and let you know.

**Comment by wgy8283335** [comment]
@ThanoshanMV There are some conflicts that need to be resolved.

**Comment by ThanoshanMV** [comment]
I resolved them @wgy8283335

**Review by wgy8283335 — APPROVED** [review]

**Pull request merged**